Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -406,6 +409,25 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         cell.imageView.contentMode = UIViewContentMode.Center
     }
 
+    private func setBlurredBackground(image: UIImage, withURL url: NSURL, forCell cell: ThumbnailCell) {
+        let blurredKey = "\(url.absoluteString)!blurred"
+        if let blurredImage = SDImageCache.sharedImageCache().imageFromMemoryCacheForKey(blurredKey) {
+            cell.backgroundImage.image = blurredImage
+        } else {
+            dispatch_async(self.blurQueue) {
+                let blurredImage = image.applyLightEffect()
```

Comment:
The `UIKit` API isn't thread-safe, so there's no guarantee this will always work as expected. We'll *probably* be OK, but it's worth keeping any eye out for any weird side effects or regressions for top site images.

Just curious: I noticed you pinned down the blur as one of causes of bad top sites perf -- how much of a difference does this PR make?